policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -425,7 +442,89 @@ impl ServiceRoutes {
                 policy.opaque = self.opaque;
                 modified = true;
             }
+            if self.accrual != policy.accrual {
+                policy.accrual = self.accrual;
+                modified = true;
+            }
             modified
         });
     }
 }
+
+fn parse_accrual_config(
+    annotations: &std::collections::BTreeMap<String, String>,
+) -> Result<Option<FailureAccrual>> {
+    annotations
+        .get("balancer.linkerd.io/failure-accrual")
+        .map(|mode| {
+            if mode == "consecutive" {
+                let max_failures = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-max-failures")
+                    .map(|s| s.parse::<u32>())
+                    .transpose()?
+                    .unwrap_or(7);
+
+                let max_penalty = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-max-penalty")
+                    .map(|s| parse_duration(s))
+                    .transpose()?
+                    .unwrap_or_else(|| time::Duration::from_secs(60));
+
+                let min_penalty = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-min-penalty")
+                    .map(|s| parse_duration(s))
+                    .transpose()?
+                    .unwrap_or_else(|| time::Duration::from_secs(1));
+                let jitter = annotations
+                    .get("balancer.linkerd.io/failure-accrual-consecutive-jitter-ratio")
+                    .map(|s| s.parse::<f32>())
+                    .transpose()?
+                    .unwrap_or(0.5);
+                if min_penalty > max_penalty {
+                    bail!(
+                        "min_penalty ({min_penalty:?}) cannot exceed max_penalty ({max_penalty:?})",
+                    );
+                }
```

Comment:
not a blocker, but potentially worth a follow-up change: proxies will do a bunch of additional validation for exponential backoff parameters: https://github.com/linkerd/linkerd2-proxy/blob/846ce1b5243fcba8958414ae732156f3b1639747/linkerd/exp-backoff/src/lib.rs#L56-L70

it might be worth doing the same validation in the controller, if that allows us to better surface configuration errors to the user?

Reply:
Yeah, great idea. I'll do a follow-up PR for this.